### ISSUE TITLE
Fedora 32 - Error while trying to install Streamdeck-ui

### ISSUE THREAD

**Issue by Jonathaan44** [timothycrosley/streamdeck-ui#55]
I don't understand what exacting what I am missing to install Streamdeck-ui.
Any suggestions?? 

Thanks!

Logs: 
```
$ sh fedora_install.sh
Installing libraries
Last metadata expiration check: 0:38:43 ago on Thu 11 Jun 2020 03:47:23 PM EDT.
Package python3-devel-3.8.3-1.fc32.x86_64 is already installed.
Package libusb-devel-1:0.1.5-16.fc32.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
Adding udev rules and reloading
usermod: group 'plugdev' does not exist
SUBSYSTEM=="usb", ATTRS{idVendor}=="0fd9", ATTRS{idProduct}=="0060", MODE:="666", GROUP="plugdev"
SUBSYSTEM=="usb", ATTRS{idVendor}=="0fd9", ATTRS{idProduct}=="0063", MODE:="666", GROUP="plugdev"
SUBSYSTEM=="usb", ATTRS{idVendor}=="0fd9", ATTRS{idProduct}=="006c", MODE:="666", GROUP="plugdev"
SUBSYSTEM=="usb", ATTRS{idVendor}=="0fd9", ATTRS{idProduct}=="006d", MODE:="666", GROUP="plugdev"
Unplug and replug in device for the new udev rules to take effect
Installing streamdeck_ui
Collecting streamdeck_ui
  Using cached streamdeck_ui-1.0.2-py3-none-any.whl (1.2 MB)
Requirement already satisfied: pyside2<6.0,>=5.13 in /home/Jonathaan44/.local/lib/python3.8/site-packages (from streamdeck_ui) (5.14.2.1)
Collecting hidapi<0.8.0,>=0.7.99
  Using cached hidapi-0.7.99.post21.tar.gz (51 kB)
Requirement already satisfied: streamdeck<0.7.0,>=0.6.3 in /home/Jonathaan44/.local/lib/python3.8/site-packages (from streamdeck_ui) (0.6.3)
Requirement already satisfied: pillow<7.0,>=6.1 in /home/Jonathaan44/.local/lib/python3.8/site-packages (from streamdeck_ui) (6.2.2)
Requirement already satisfied: pynput<2.0,>=1.4 in /home/Jonathaan44/.local/lib/python3.8/site-packages (from streamdeck_ui) (1.6.8)
Requirement already satisfied: shiboken2==5.14.2.1 in /home/Jonathaan44/.local/lib/python3.8/site-packages (from pyside2<6.0,>=5.13->streamdeck_ui) (5.14.2.1)
Requirement already satisfied: setuptools>=19.0 in /usr/lib/python3.8/site-packages (from hidapi<0.8.0,>=0.7.99->streamdeck_ui) (41.6.0)
Requirement already satisfied: python-xlib>=0.17; "linux" in sys_platform in /home/Jonathaan44/.local/lib/python3.8/site-packages (from pynput<2.0,>=1.4->streamdeck_ui) (0.27)
Requirement already satisfied: six in /usr/lib/python3.8/site-packages (from pynput<2.0,>=1.4->streamdeck_ui) (1.14.0)
Building wheels for collected packages: hidapi
  Building wheel for hidapi (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-xook849b/hidapi/setup.py'"'"'; __file__='"'"'/tmp/pip-install-xook849b/hidapi/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-n_5p5qdj
       cwd: /tmp/pip-install-xook849b/hidapi/
  Complete output (21 lines):
  running bdist_wheel
  running build
  running build_ext
  skipping 'hid.c' Cython extension (up-to-date)
  skipping 'hidraw.c' Cython extension (up-to-date)
  building 'hid' extension
  creating build
  creating build/temp.linux-x86_64-3.8
  creating build/temp.linux-x86_64-3.8/hidapi
  creating build/temp.linux-x86_64-3.8/hidapi/libusb
  gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ihidapi/hidapi -I/usr/include/libusb-1.0 -I/usr/include/python3.8 -c hid.c -o build/temp.linux-x86_64-3.8/hid.o
  gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ihidapi/hidapi -I/usr/include/libusb-1.0 -I/usr/include/python3.8 -c hidapi/libusb/hid.c -o build/temp.linux-x86_64-3.8/hidapi/libusb/hid.o
  hidapi/libusb/hid.c:26: warning: "_GNU_SOURCE" redefined
     26 | #define _GNU_SOURCE /* needed for wcsdup() before glibc 2.10 */
        |
  <command-line>: note: this is the location of the previous definition
  creating build/lib.linux-x86_64-3.8
  gcc -pthread -shared -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -g -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -g build/temp.linux-x86_64-3.8/hid.o build/temp.linux-x86_64-3.8/hidapi/libusb/hid.o -L/usr/lib64 -lusb-1.0 -ludev -lrt -o build/lib.linux-x86_64-3.8/hid.cpython-38-x86_64-linux-gnu.so
  /usr/bin/ld: cannot find -ludev
  collect2: error: ld returned 1 exit status
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for hidapi
  Running setup.py clean for hidapi
Failed to build hidapi
Installing collected packages: hidapi, streamdeck-ui
    Running setup.py install for hidapi ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-xook849b/hidapi/setup.py'"'"'; __file__='"'"'/tmp/pip-install-xook849b/hidapi/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-m2d2tncb/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/Jonathaan44/.local/include/python3.8/hidapi
         cwd: /tmp/pip-install-xook849b/hidapi/
    Complete output (21 lines):
    running install
    running build
    running build_ext
    skipping 'hid.c' Cython extension (up-to-date)
    skipping 'hidraw.c' Cython extension (up-to-date)
    building 'hid' extension
    creating build
    creating build/temp.linux-x86_64-3.8
    creating build/temp.linux-x86_64-3.8/hidapi
    creating build/temp.linux-x86_64-3.8/hidapi/libusb
    gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ihidapi/hidapi -I/usr/include/libusb-1.0 -I/usr/include/python3.8 -c hid.c -o build/temp.linux-x86_64-3.8/hid.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Ihidapi/hidapi -I/usr/include/libusb-1.0 -I/usr/include/python3.8 -c hidapi/libusb/hid.c -o build/temp.linux-x86_64-3.8/hidapi/libusb/hid.o
    hidapi/libusb/hid.c:26: warning: "_GNU_SOURCE" redefined
       26 | #define _GNU_SOURCE /* needed for wcsdup() before glibc 2.10 */
          |
    <command-line>: note: this is the location of the previous definition
    creating build/lib.linux-x86_64-3.8
    gcc -pthread -shared -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -g -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -g build/temp.linux-x86_64-3.8/hid.o build/temp.linux-x86_64-3.8/hidapi/libusb/hid.o -L/usr/lib64 -lusb-1.0 -ludev -lrt -o build/lib.linux-x86_64-3.8/hid.cpython-38-x86_64-linux-gnu.so
    /usr/bin/ld: cannot find -ludev
    collect2: error: ld returned 1 exit status
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-xook849b/hidapi/setup.py'"'"'; __file__='"'"'/tmp/pip-install-xook849b/hidapi/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-m2d2tncb/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/Jonathaan44/.local/include/python3.8/hidapi Check the logs for full command output.

```

**Comment by devrimgunduz** [comment]
dnf install systemd-devel

**Comment by Cilyan** [comment]
You will need libjpeg-turbo-devel too, as Pillow will need it to compile. (Why is version 7.2 not supported ?)

**Comment by dodgyrabbit** [comment]
Sharing some of my own observations:
The currently released version (1.0.2) references some older python libraries and on Fedora 33 the installer cannot simply resolve them and tries to compile locally. This leads to other problems when trying to compile (building pillow is not simple). As pointed out, Pillow (6.1) is one of the problematic ones.

The latest code in master, Pillow 8.1 is used.

hidapi is also something that it appears is needed to work on Fedora.

I've tested from a completely fresh install on Fedora that one can install the dependencies, install poetry, use it to install the python modules and run without problems.

I'm closing this issue and we can review again once 1.0.3 has been published.
